src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -862,6 +870,12 @@ WorkerShardStats(ShardPlacement *placement, Oid relationId, const char *shardNam
 	MultiConnection *connection = GetPlacementConnection(connectionFlags, placement,
 														 NULL);
 
+	/*
+	 * This code-path doesn't support optional connections, so we don't expect
+	 * NULL connections.
+	 */
+	Assert(connection != NULL && (connectionFlags & OPTIONAL_CONNECTION) == 0);
```

Comment:
these asserts were added mostly because of the Microsoft static code analyzers not to complain. 
(I was not sure whether to add, but added to prevent those tools fails because on 9.3 I remember we had to add those asserts https://github.com/citusdata/citus/pull/3805/files)

Reply:
Left the assertion as `Assert(connection != NULL)`